File: models/Game.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const GameSchema = new Schema({
    userId: {
        type: mongoose.Types.ObjectId,
        required: true
    },
    participants: [mongoose.Types.ObjectId],
    name: {
        type: String,
        required: true
    },
    distance: {
        type: Number,
        default: 0.0
    },
    progress: {
        distance: {
            type: Number,
            default: 0.0
        },
        days: {
            type: Number,
            default: 0
        },
        lastDay: {
            type: Date,
            default: Date.now
        }
    },
    started_at: {
        type: Date,
        default: Date.now
    },
    status: {
        type: Number,
        default: 0 //0 is running, 1 is success, 2 is fail
    },
});

mongoose.model('games', GameSchema);

File: app/src/main/java/com/swordrunner/swordrunner/data/model/Game.java
package com.swordrunner.swordrunner.data.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Game {

    @SerializedName("_id")
    public String id = null;

    @SerializedName("user_id")
    String userId = null;

    public List<String> participants = new ArrayList<>();

    public String name = "New game";

    public double distance = 1.0;

    public Progress progress = new Progress();


    public class Progress {
        public int days = 0;
        public double distance = 0.0;
        public Date lastDay = null;
    }
    public Game(){

    }
    public Game(String id, String userId, List<String> participants, String name, double distance, Progress progress) {
        this.id = id;
        this.userId = userId;
        this.participants = participants;
        this.name = name;
        this.distance = distance;
        this.progress = progress;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }
}